gammapy/scripts/analysis.py

```diff
@@ -51,6 +63,118 @@ def settings(self):
         """Configuration settings for the analysis session."""
         return self.config.settings
 
+    def get_observations(self):
+        """Fetch observations from the data store according to criteria defined in the configuration."""
+        self.config.validate()
+
+        datastore_path = make_path(self.settings["observations"]["datastore"])
+        if datastore_path.is_file():
+            datastore = DataStore().from_file(datastore_path)
+        elif datastore_path.is_dir():
+            datastore = DataStore().from_dir(datastore_path)
+        else:
+            raise FileNotFoundError("Datastore {} not found.".format(datastore_path))
+
+        ids = set()
+        selection = dict()
+        for criteria in self.settings["observations"]["filters"]:
+            selected_obs = ObservationTable()
+
+            # -- TODO Handled by datastore.obs_table.select_observations
+            # -
+            selection["type"] = criteria["filter_type"]
+            for key, val in criteria.items():
+                if key in ["lon", "lat", "radius", "border"]:
+                    val = Angle(val)
+                selection[key] = val
+            if selection["type"] == "angle_box":
+                selection["type"] = "par_box"
+                selection["value_range"] = Angle(criteria["value_range"])
+            if selection["type"] == "sky_circle" or selection["type"].endswith("_box"):
+                selected_obs = datastore.obs_table.select_observations(selection)
+            if selection["type"] == "par_value":
+                mask = (
+                    datastore.obs_table[criteria["variable"]] == criteria["par_value"]
+                )
+                selected_obs = datastore.obs_table[mask]
+            if selection["type"] == "ids":
+                obs_list = datastore.get_observations(criteria["obs_ids"])
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if selection["type"] == "all":
+                obs_list = datastore.get_observations()
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            # --
+            # -
+
+            if len(selected_obs):
+                if "exclude" in criteria and criteria["exclude"]:
+                    ids.difference_update(selected_obs["OBS_ID"].tolist())
+                else:
+                    ids.update(selected_obs["OBS_ID"].tolist())
+        if len(ids):
```

Comment:
Can you remove the if here?
`datastore.get_observations` should just be able to handle the empty list case.
If it doesn't already, make it so. But probably it already does.